javaslang/src/main/java/javaslang/collection/Stack.java

```diff
@@ -38,7 +39,7 @@
  * @author Daniel Dietrich
  * @since 2.0.0
  */
-public interface Stack<T> extends LinearSeq<T> {
+public interface Stack<T> extends Kind1<List<?>, T>, LinearSeq<T> {
```

Comment:
I will make Stack an ordinary interface having only a few Stack-related methods and which don't extends LinearSeq. A few people discussed about that on gitter and I made a poll. Scala has removed it entirely but having a simple interface should be ok. I will take care of the javadoc in order not to confuse users whether to to use prepend or push on a List...